src/sentry/static/sentry/app/components/events/interfaces/frameRegisters.jsx

```diff
@@ -0,0 +1,139 @@
+import {Flex, Box} from 'grid-emotion';
+import PropTypes from 'prop-types';
+import React from 'react';
+import styled from 'react-emotion';
+
+import Tooltip from 'app/components/tooltip';
+import {t} from 'app/locale';
+import {defined, objectToArray} from 'app/utils';
+
+const REGISTER_VIEWS = [t('Hexadecimal'), t('Numeric')];
+
+export class RegisterValue extends React.Component {
+  static propTypes = {
+    value: PropTypes.oneOfType([PropTypes.string, PropTypes.number]).isRequired,
+  };
+
+  constructor(props) {
+    super(props);
+    this.state = {
+      view: 0,
+    };
+  }
+
+  toggleView = () => {
+    this.setState({
```

Comment:
prefer state callback style when updating state with previous state.

e.g. `this.setState(state => ({ view: state.view + 1 }))`